envoy/buffer/buffer.h

```diff
@@ -458,6 +458,40 @@ class Instance {
     writeInt<ByteOrder::BigEndian, T, Size>(value);
   }
 
+  /**
+   * Copy multiple arguments to the buffer. The type of arguments must be automatically convertible
```

Comment:
Nit: remove "The type of arguments..." sentence from here, but add the detail about which types are convertible into the description for 'args'. No need to duplicate the concept between the general description and the args -- just put all the detail under 'args'.